src/axeResultToEarl.ts

```diff
@@ -0,0 +1,48 @@
+import { RawResult, Assertion, TestResult } from './types'
+import testResults from './testResult'
+
+export default function axeResultToEarl(
+  rawResults: RawResult[]
+): Assertion[] {
+  return rawResults.reduce(
+    (asserts: Assertion[], axeResult: RawResult): Assertion[] => {
+      let results: TestResult[] = []
+      results = results.concat(
+        testResults(axeResult, axeResult.violations)
+      )
+      results = results.concat(
+        testResults(axeResult, axeResult.passes)
+      )
+      results = results.concat(
+        testResults(axeResult, axeResult.incomplete)
+      )
+
+      if (results.length === 0) {
+        results.push({
+          '@type': 'TestResult',
+          outcome: 'earl:inapplicable'
+        })
+      }
+
+      const newAsserts = results.map(result => {
+        const match = axeResult.helpUrl.match(
+          /axe\/([1-9][0-9]*\.[1-9][0-9]*)\//
+        )
+        const version = (match && match[1]) || ''
+        return {
+          '@type': 'Assertion',
+          mode: 'earl:automatic',
+          assertedBy: `https://github.com/dequelabs/axe-core/releases/tag/${version}.0`,
+          test: {
+            '@type': 'TestCase',
+            '@id': axeResult.helpUrl
+          },
+          result
+        } as Assertion
```

Comment:
Instead of type casting here, we ought to declare the return type of this function. This produces more self-documenting code.

NOTE: type casting is almost never the "correct" answer.

Reply:
Probably isn't. Couldn't figure out how to solve it differently. There's probably something I'm doing wrong in my type declaration, just not sure what.

Reply:
Extract the `function` of `map` with a return type of `Assertion`, also perhaps the return object can be assigned to a variable of expected type, to seek `ts` help to ascertain the right object type (ts will help saying certain member in the object is missing etc).

```js
const result: Assertion = { ... } // ts should help construct the right type of object.
```

Doing `return <Assertion> { ... }` will also help construct the object, rather than using the `as Assertion`, which will force a coercion.